### DISCUSSION TITLE
Adding support for Laravel Octane?

### DISCUSSION THREAD

**Discussion by sikhlana** [NativePHP/laravel#416] [Ideas]
If my understanding is correct, even the production built NativePHP uses PHP's development server to serve the HTTP requests to the application.

Maybe add support for `laravel/octane` + `swoole` (may not have Windows support) / `roadrunner` for the HTTP server instead?

**Comment by simonhamp** [comment]
I would love to use Octane, but it relies on the `pcntl` extension, which means it won't work on Windows.

We'll need our own custom implementation that doesn't rely on `pcntl`.

But yeh, this would be great.

<details><summary>Using the development server</summary>
<p>

Yes, it's true, even apps that go to production will use PHP's built-in dev server. So far this isn't an issue, it merely acts as a lightweight interface to running the PHP code. But for sure I would like a more robust solution. Ideally where we're not having to listen on a network interface at all.

</p>
</details>

**Reply by sikhlana** [reply]
Just an idea for HTTP replacement: maybe we can create some kind of IPC bridge between `electron` and `php` just like how it's done in `roadrunner`: https://github.com/roadrunner-php/goridge (IPC between `golang` and `php`).

I will give this approach a try when I have some free time.

**Reply by yangweijie** [reply]
In windows you can run swoole in cygwin 。 which pcntl and other linux only extension can be built and run well。I build php7.4 with swoole 4.x and ffi successfully.
